drivers/SmartThings/zwave-switch/src/multi-metering-switch/init.lua

```diff
@@ -0,0 +1,182 @@
+--
+-- Licensed under the Apache License, Version 2.0 (the "License");
+-- you may not use this file except in compliance with the License.
+-- You may obtain a copy of the License at
+--
+--     http://www.apache.org/licenses/LICENSE-2.0
+--
+-- Unless required by applicable law or agreed to in writing, software
+-- distributed under the License is distributed on an "AS IS" BASIS,
+-- WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+-- See the License for the specific language governing permissions and
+-- limitations under the License.
+
+local st_device = require "st.device"
+local capabilities = require "st.capabilities"
+--- @type st.zwave.CommandClass
+local cc = require "st.zwave.CommandClass"
+--- @type st.zwave.CommandClass.Meter
+local Meter = (require "st.zwave.CommandClass.Meter")({version = 3})
+--- @type st.zwave.CommandClass.SwitchBinary
+local SwitchBinary = (require "st.zwave.CommandClass.SwitchBinary")({version = 1})
+local Basic = (require "st.zwave.CommandClass.Basic")({version = 1})
+
+local MULTI_METERING_SWITCH_FINGERPRINTS = {
+  {mfr = 0x0086, prod = 0x0003, model = 0x0084, children = 2}, -- Aeotec Nano Switch 1
+  {mfr = 0x0086, prod = 0x0103, model = 0x0084, children = 2}, -- Aeotec Nano Switch 1
+  {mfr = 0x0086, prod = 0x0203, model = 0x0084, children = 2}, -- AU Aeotec Nano Switch 1
+  {mfr = 0x027A, prod = 0xA000, model = 0xA003, children = 2}, -- Zooz Double Plug 1
+  {mfr = 0x027A, prod = 0xA000, model = 0xA004, children = 5}, -- Zooz ZEN Power Strip 1
+  {mfr = 0x015F, prod = 0x3102, model = 0x0201, children = 1}, -- WYFY Touch 1-button Switch
+  {mfr = 0x015F, prod = 0x3102, model = 0x0202, children = 2}, -- WYFY Touch 2-button Switch
+  {mfr = 0x015F, prod = 0x3102, model = 0x0204, children = 4}, -- WYFY Touch 4-button Switch
+  {mfr = 0x015F, prod = 0x3111, model = 0x5102, children = 1}, -- WYFY Touch 1-button Switch
+  {mfr = 0x015F, prod = 0x3121, model = 0x5102, children = 2}, -- WYFY Touch 2-button Switch
+  {mfr = 0x015F, prod = 0x3141, model = 0x5102, children = 4}, -- WYFY Touch 4-button Switch
+}
+
+local function can_handle_multi_metering_switch(opts, driver, device, ...)
+  for _, fingerprint in ipairs(MULTI_METERING_SWITCH_FINGERPRINTS) do
+    if device:id_match(fingerprint.mfr, fingerprint.prod, fingerprint.model) then
+      return true
+    end
+  end
+  return false
+end
+
+local function get_children_amount(device)
+  for _, fingerprint in ipairs(MULTI_METERING_SWITCH_FINGERPRINTS) do
+    if device:id_match(fingerprint.mfr, fingerprint.prod, fingerprint.model) then
+      return fingerprint.children
+    end
+  end
+end
+
+local function get_profile(device)
+  if device:get_manufacturer() == 0x015F then
+    return "switch-binary"
+  else
+    return "metering-switch"
+  end
+end
+
+local function device_added(driver, device, event)
+  if device.network_type == st_device.NETWORK_TYPE_ZWAVE then
+    local children_amount = get_children_amount(device)
```

Comment:
@NikitaKuchta You may utilize similar mechanism to this one:  https://github.com/SmartThingsCommunity/SmartThingsEdgeDrivers/blob/main/drivers/SmartThings/zwave-switch/src/zwave-dual-switch/dual_switch_configurations.lua
to obtain all needed config information (like: profile and number of child devices) and avoid 'if manufacturer' statements in the code.